pkg/noderesourcetopology/filter.go

```diff
@@ -89,16 +114,26 @@ func resourcesAvailableInAnyNUMANodes(logKey string, numaNodes NUMANodeList, res
 
 			resourceBitmask.Add(numaNode.NUMAID)
 			klog.V(6).InfoS("feasible", "logKey", logKey, "node", nodeName, "NUMA", numaNode.NUMAID, "resource", resource)
+
+			// according to TopologyManager, the preferred NUMA affinity, is the narrowest one.
+			// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/topologymanager/policy.go#L155-L157
+			// in single-numa-node policy it's very simple. since all resources should be allocated from a single NUMA,
+			// it means that the lowest NUMA ID (with available resources) is the one to be selected by Kubelet.
+			if numaNode.NUMAID < numaID {
+				numaID = numaNode.NUMAID
+			}
 		}
 		bitmask.And(resourceBitmask)
 		if bitmask.IsEmpty() {
 			klog.V(5).InfoS("early verdict", "logKey", logKey, "node", nodeName, "resource", resource, "suitable", "false")
-			return false
+			return numaID, false
 		}
 	}
-	ret := !bitmask.IsEmpty()
+
+	// at least one NUMA node is available
+	ret := bitmask.Count() > 0
```

Comment:
And difference with `bitmask.IsEmpty()`? If no, seems `bitmask.IsEmpty()` is more efficient.

Reply:
Done.